lib/screens/all_apps.dart

```diff
@@ -58,39 +58,95 @@ class AppsScreeen extends StatelessWidget {
   }
 }
 
-class AppWIdget extends StatelessWidget {
-  final apps = Get.put(MyApps());
+class AppWIdget extends StatefulWidget {
+  @override
+  _AppWIdgetState createState() => _AppWIdgetState();
+}
+
+class _AppWIdgetState extends State<AppWIdget> {
+  List<dynamic> displayApps = Get.put(MyApps()).apps;
+  final TextEditingController _searchController = TextEditingController();
+
+  // search bar on the top
+  Widget buildSearchBar() {
+    return Padding(
+      padding: EdgeInsets.fromLTRB(40, 8, 8, 8),
+      child: Container(
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(5.0),
+          color: Colors.grey.withOpacity(0.6),
+        ),
+        child: TextField(
+          decoration: InputDecoration(
+            disabledBorder: InputBorder.none,
+            hintText: "Search",
+            contentPadding: EdgeInsets.only(left: 10.0),
+          ),
+          controller: _searchController,
+          onChanged: (value) {
+            _upDateAppList(query: _searchController.text);
+          },
+        ),
+      ),
+    );
+  }
+
+  // this function will update the displayApps list with the searched apps only
+  void _upDateAppList({String query}) {
+    query = query.trim().toLowerCase();
+    var _allApps = Get.put(MyApps()).apps;
+    var _reqApp = [];
+    _allApps.forEach((app) {
+      if (app['label'].toString().toLowerCase().contains(query)) {
```

Comment:
lets have a searchcontroller (i.e GetController ) which have this method and we should be able to change the algorithm there